test/e2e/e2e_local_queue_defaulting_test.go

```diff
@@ -230,4 +195,120 @@ var _ = Describe("LocalQueueDefaulting", Ordered, func() {
+			Expect(err).NotTo(HaveOccurred())
+			Expect(createdPodWithoutQueue.Labels).NotTo(HaveKey(testutils.QueueLabel))
+			Eventually(func() bool {
+				if !testutils.IsPodScheduled(ctx, kubeClient, ns.Name, createdPodWithoutQueue.Name) {
+					return false
+				}
+				updatedPod, err := kubeClient.CoreV1().Pods(ns.Name).Get(ctx, createdPodWithoutQueue.Name, metav1.GetOptions{})
+				if err != nil {
+					return false
+				}
+				return updatedPod.Status.Phase == corev1.PodRunning
+			}, testutils.OperatorReadyTime, testutils.OperatorPoll).Should(BeTrue(), "Pod did not reach expected state (Sucessfully Scheduled and currently Running)")
+			time.Sleep(5 * time.Minute)
+			//Clean Up - resources deprovision
+			deleteNamespace(context.TODO(), ns)
+			deleteClusterQueueAndResourceFlavor(kueueClient, initialKueueInstance)
+		})
+	})
 })
+
+func deleteClusterQueueAndResourceFlavor(kueueClient *upstreamkueueclient.Clientset, initialKueueInstance *kueueoperatorv1.Kueue) {
+	By("deleting cluster queue")
+	err := kueueClient.KueueV1beta1().ClusterQueues().Delete(context.TODO(), "test-clusterqueue", metav1.DeleteOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	Eventually(func() error {
+		_, err := kueueClient.KueueV1beta1().ClusterQueues().Get(context.TODO(), "test-clusterqueue", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("ClusterQueue test-clusterqueue still exists")
+	}, testutils.DeletionTime, testutils.DeletionPoll).Should(Succeed(), "ClusterQueue was not cleaned up properly")
+	By("deleting resource flavor")
+	err = kueueClient.KueueV1beta1().ResourceFlavors().Delete(context.TODO(), "default", metav1.DeleteOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	Eventually(func() error {
+		_, err := kueueClient.KueueV1beta1().ResourceFlavors().Get(context.TODO(), "default", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("ResourceFlavor default still exists")
+	}, testutils.DeletionTime, testutils.DeletionPoll).Should(Succeed(), "ResourceFlavor was not cleaned up properly")
+	applyKueueConfig(context.TODO(), initialKueueInstance.Spec.Config, kubeClient)
+}
+
+func deleteNamespace(ctx context.Context, namespace *corev1.Namespace) {
+	By(fmt.Sprintf("Deleting namespace %s", namespace.Name))
+	err := kubeClient.CoreV1().Namespaces().Delete(ctx, namespace.Name, metav1.DeleteOptions{})
+	Expect(err).NotTo(HaveOccurred())
+	testutils.WaitForAllPodsInNamespaceDeleted(ctx, clients.GenericClient, namespace)
+}
+
+func createClusterQueueAndResourceFlavor(ctx context.Context, labelPolicy string) {
+	var err error
+	kueueInstance, err := clients.KueueClient.KueueV1().Kueues().Get(ctx, "cluster", metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred(), "Failed to fetch Kueue instance")
+	initialKueueInstance = kueueInstance.DeepCopy()
+	if labelPolicy != "" {
+		kueueInstance.Spec.Config.WorkloadManagement.LabelPolicy = kueueoperatorv1.LabelPolicy(labelPolicy)
```

Comment:
It's worth extracting the configuration of labelPolicy from this function as it doesn't reflect the function name